src/main/java/de/tum/in/www1/artemis/service/ResultService.java

```diff
@@ -217,12 +215,10 @@ public Result addFeedbackToResult(@NotNull Result result, List<Feedback> feedbac
     public List<Feedback> getFeedbacksForResult(Result result) {
         this.filterSensitiveInformationIfNecessary(result.getParticipation(), result);
 
-        // remove unnecessary data to keep the json payload smaller
-        for (Feedback feedback : result.getFeedbacks()) {
-            feedback.setResult(null);
-        }
-
-        return result.getFeedbacks();
+        return result.getFeedbacks().stream() //
+                .map(feedback -> feedback.result(null)) // remove unnecessary data to keep the json payload smaller
+                .sorted(Comparator.comparing(feedback -> Objects.requireNonNullElse(feedback.getType(), FeedbackType.AUTOMATIC))) // sort according to FeedbackType enum order.
```

Comment:
`feedback.getType()` should never be `null`. However we code defensively and put those cases at the end of the order